### PR TITLE
Setup vsock devices for APPVMs and use it for Waypipe

### PR DESCRIPTION
**Pull request by nesteroff**
This PR makes Waypipe work over VSOCK instead of UNIX domain sockets. It increases performance by getting rid of the unnecessary overhead caused by forwarding UNIX sockets over SSH. Testing with iperf show that UNIX sockets over SSH can transfer about 4 Gbits/sec on Lenovo X1 Carbon but with VSOCK it can handle about 15 Gbits/sec between GUIVM and APPVMs.

### CODE REVIEW

File: targets/lenovo-x1-carbon.nix [hunk 1]
```diff
@@ -120,6 +120,7 @@
                     macAddress = "02:00:00:03:03:05";
                     ramMb = 3072;
                     cores = 4;
+                    cid = 4;
```
Comment (brianmcgillion) [inline targets/lenovo-x1-carbon.nix:123]: is there anyway to script this as e.g. ++cid so as to avoid hard coding for each VM?

Comment (nesteroff) [reply targets/lenovo-x1-carbon.nix:123]: Sure, I'll check how to do it.

**Comment by vadika** [comment]
It's basically bad idea cause we need to know this addresses exactly for each VM to be able to address vsock endpoints. —vadik ***@***.*** 8. Sep 2023, at 13.27, Brian McGillion ***@***.***> wrote:﻿
@brianmcgillion commented on this pull request.



In targets/lenovo-x1-carbon.nix:
> @@ -120,6 +120,7 @@
                     macAddress = "02:00:00:03:03:05";
                     ramMb = 3072;
                     cores = 4;
+                    cid = 4;

is there anyway to script this as e.g. ++cid so as to avoid hard coding for each VM?

**Comment by nesteroff** [comment]
In this particular case APPVMs connect to GUIVM but not the other way around. It means that they only need to know the CID of the GUIVM which is always 3 (first available number).

File: targets/lenovo-x1-carbon.nix (hunk 1 above)
Comment (nesteroff) [reply targets/lenovo-x1-carbon.nix]: Added auto-assigning but at the same time kept the ability to override with a static value.

File: user-apps/vsockproxy/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  lib,
+  ...
+}:
+stdenv.mkDerivation {
+  name = "vsockproxy";
+
+  nativeBuildInputs = with pkgs; [meson ninja];
+
+  src = pkgs.fetchFromGitHub {
+    owner = "nesteroff";
```
Comment (mikatammi) [inline user-apps/vsockproxy/default.nix:15]: I think we should create a repo inside TIIUAE for this

**Review by mikatammi — CHANGES_REQUESTED** [review]
Please create repo under `tiiuae` for `vsockproxy`. Otherwise LGTM

File: user-apps/vsockproxy/default.nix (hunk 2 above)
Comment (nesteroff) [reply user-apps/vsockproxy/default.nix]: Ok, now it's in tiiuae.

**Comment by mikatammi** [comment]
Needs rebase now

File: modules/virtualization/microvm/appvm.nix [hunk 3]
```diff
@@ -64,11 +71,21 @@
                 mac = vm.macAddress;
               }
             ];
-            # Use qboot BIOS on x86_64-linux as workaround
-            qemu.extraArgs = lib.optionals (config.nixpkgs.hostPlatform.system == "x86_64-linux") [
-              "-bios"
-              "${pkgs.qboot}/bios.bin"
-            ];
+            qemu.extraArgs =
```
Comment (mikatammi) [inline modules/virtualization/microvm/appvm.nix:74]: No need for this stuff anymore, as the machine type has now changed back to q35

**Review by mikatammi — APPROVED** [review]
LGTM. I will test this first

**Pull request merged**